src/Riok.Mapperly/Descriptors/WellKnownTypes.cs

```diff
@@ -9,6 +9,7 @@ namespace Riok.Mapperly.Descriptors;
 public class WellKnownTypes
 {
     private readonly Compilation _compilation;
+    private readonly Dictionary<string, INamedTypeSymbol?> _cachedTypes = new();
 
     private INamedTypeSymbol? _referenceHandlerAttribute;
```

Comment:
I think I'd remove all those fields and always use `GetTypeSymbol` as it should be constant time and pretty fast now and the fields grew a lot over the past.

Reply:
Okay I'll replace the fields with `GetTypeSymbol(typeof(ImmutableList<>))` etc.
Would it be worth keeping the heavily used ones like `ICollectionT`?